### PR TITLE
#203 Commit

### LINKED ISSUE #203: Set starting offsets earliest when checkpoint folder is empty
**Situation**
Currently, `startingOffsets` is set to `earliest` by `KafkaStreamReader` if the checkpoint folder doesn't exist, so the query ingests from the beginning at the first microbatch.
However, `earliest` is not set if the already folder exists, but is empty. This is inconsistent behaviour and should be fixed.
https://github.com/AbsaOSS/hyperdrive/blob/develop/ingestor-default/src/main/scala/za/co/absa/hyperdrive/ingestor/implementation/reader/kafka/KafkaStreamReader.scala#L98

**Desired outcome**
If the checkpoint folder exists, but is empty, `startingOffsets` should be set to `earliest`

### PR DESCRIPTION
**Pull request by AshishTiwary**
Closes #203

### CODE REVIEW

File: driver/src/main/scala/za/co/absa/hyperdrive/driver/drivers/PropertiesIngestionDriver.scala [hunk 1]
```diff
@@ -33,6 +33,9 @@ object PropertiesIngestionDriver extends IngestionDriver {
   private val logger = LogManager.getLogger
 
   def main(args: Array[String]): Unit = {
+
+    //System.setProperty("hadoop.home.dir", "C:\\winutils\\")
+
```
Comment (kevinwallimann) [inline driver/src/main/scala/za/co/absa/hyperdrive/driver/drivers/PropertiesIngestionDriver.scala]: Please remove this comment.

File: shared/src/main/scala/za/co/absa/hyperdrive/shared/utils/FileUtils.scala [hunk 2]
```diff
@@ -27,6 +27,8 @@ private[hyperdrive] object FileUtils {
 
   def notExists(file: String, configuration: Configuration): Boolean = !exists(file, configuration)
 
+
+
```
Comment (kevinwallimann) [inline shared/src/main/scala/za/co/absa/hyperdrive/shared/utils/FileUtils.scala]: Please remove this whitespace

File: shared/src/main/scala/za/co/absa/hyperdrive/shared/utils/FileUtils.scala [hunk 3]
```diff
@@ -37,7 +39,8 @@ private[hyperdrive] object FileUtils {
   def isEmpty(directory: String, configuration: Configuration): Boolean = {
     val fs = getFileSystem(configuration)
     val path = new Path(directory)
-    fs.exists(path) && !fs.listLocatedStatus(path).hasNext
+    //fs.exists(path) && !fs.listLocatedStatus(path).hasNext
```
Comment (kevinwallimann) [inline shared/src/main/scala/za/co/absa/hyperdrive/shared/utils/FileUtils.scala]: Please remove this comment

**Review by kevinwallimann — COMMENTED** [review]
Could you add this test from branch feature/203-starting-offsets-unittests as well?
https://github.com/AbsaOSS/hyperdrive/compare/feature/203-starting-offsets-unittests#diff-87b6981a1afce27526c0f5ac5251d4b2b38e3ccd99ec64cae486a14d77e571e7

File: driver/src/main/scala/za/co/absa/hyperdrive/driver/drivers/PropertiesIngestionDriver.scala (hunk 1 above)
Comment (AshishTiwary) [reply driver/src/main/scala/za/co/absa/hyperdrive/driver/drivers/PropertiesIngestionDriver.scala]: Removed it. Please check

File: shared/src/main/scala/za/co/absa/hyperdrive/shared/utils/FileUtils.scala (hunk 2 above)
Comment (AshishTiwary) [reply shared/src/main/scala/za/co/absa/hyperdrive/shared/utils/FileUtils.scala]: Removed it. Please check

File: shared/src/main/scala/za/co/absa/hyperdrive/shared/utils/FileUtils.scala (hunk 3 above)
Comment (AshishTiwary) [reply shared/src/main/scala/za/co/absa/hyperdrive/shared/utils/FileUtils.scala]: Removed it. Please check

**Review by kevinwallimann — APPROVED** [review]

**Pull request merged**